app/javascript/mastodon/actions/alerts.js

```diff
@@ -33,13 +35,18 @@ export function showAlert(title = messages.unexpectedTitle, message = messages.u
 
 export function showAlertForError(error) {
   if (error.response) {
-    const { data, status, statusText } = error.response;
+    const { data, status, statusText, headers } = error.response;
 
     if (status === 404 || status === 410) {
       // Skip these errors as they are reflected in the UI
       return { type: ALERT_NOOP };
     }
 
+    if (status === 429 && headers['x-ratelimit-reset']) {
+      const reset_date = new Date(headers['x-ratelimit-reset']);
+      return showAlert(messages.rateLimitedTitle, [messages.rateLimitedMessage, { 'retry_time': reset_date }]);
```

Comment:
The array thing seems unnecessary. Wouldn't it be just as easy to have a 3rd argument on the function and check if that argument is not undefined?

Reply:
Pushed a commit to address that, not sure how much better that is, though